### PR TITLE
Bitbucket on prem/init api

### PR DESCRIPTION
**Pull request by dan505512**
Adding queries for bitbucket on prem. Works on my machine

### CODE REVIEW

**Comment by dan505512** [comment]
RK-7017

**Comment by dan505512** [comment]
@amits1995 Fixed the api. Sending you all the queries on Slack so you can see how it looks

**Review by amits1995 — CHANGES_REQUESTED** [review]
Stricter parameters and some minors

File: src/BitBucketOnPrem.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    if (!validateUrlIsAuthorized(url)) return null;
+
+    logger.debug("Getting file content", { url, projectKey, repoName, commit, filePath });
+    const fileQuery = `${url}/rest/api/1.0/projects/${projectKey}/repos/${repoName}/browse/${filePath}?at=${commit}`;
+    const res = await fetch(fileQuery, {
+        headers: {
+            Authorization: `Bearer ${accessToken}`
+        }
+    });
+    const file = await res.json();
```
[the remark this answers is not in the document]
Comment (dan505512) [reply src/BitBucketOnPrem.ts:137]: No other way

File: src/utils.ts [hunk 2]
```diff
@@ -16,28 +18,45 @@ export const getLibraryFolder = () => {
...
-    Object.entries(settings).forEach(([key, val]) => {
-      if (store.get(overrideGlobalName(key), false)) {
-        console.info(`skipping setting ${key}:${val} since it was overriden locally`)
-        return
-      }
-      store.set(key, val)
-    });
-  }
-  return  getSettings();
+    if (!areOnPremServersValid(settings?.BitbucketOnPremServers)) {
```
Comment (amits1995) [inline src/utils.ts:41]: move to constraint directive (easily done with https://www.npmjs.com/package/graphql-constraint-directive#pattern)

Comment (amits1995) [reply src/utils.ts:41]: 2. no need to always override the servers settings

**Review by amits1995 — CHANGES_REQUESTED** [review]

File: src/BitBucketOnPrem.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (amits1995) [reply src/BitBucketOnPrem.ts:158]: sad reacts only

**Review by amits1995 — DISMISSED** [review]

**Review by amits1995 — APPROVED** [review]

**Pull request merged**